### PR TITLE
ENH: include slightly tweaked DesignerDisplay from atef

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Include atef's designer display as a general util

## Motivation and Context

I wanted to use this in another project

## How Has This Been Tested?

I'll run with this for my other project and see how it goes

## Where Has This Been Documented?

Not yet

### CODE REVIEW

**Comment by klauer** [comment]
I've copied/modified this a few times at this point. I really like the pattern.

There may be some things to borrow from here as well:
https://github.com/pcdshub/las-dispersion-scan/blob/0049619f49e7f0b5f84c6aaa9dd7fd7a252a49a7/las_dispersion_scan/widgets.py#L24-L74

**Comment by ZLLentz** [comment]
I'm going to try to finish this PR and make a tag so I can use it in https://github.com/pcdshub/pmpsdb_client

**Comment by ZLLentz** [comment]
- Added your improvements from las-dispersion-scan
- Generalized the ui directory for use in other repos
- Added simple unit tests

File: pcdsutils/qt/designer_display.py [hunk 1]
```diff
@@ -45,5 +62,7 @@ def show_type_hints(self):
                 ...
             else:
                 if obj is not None:
-                    print(f"{attr}: {obj.__class__.__module__}.{obj.__class__.__name__}")
-
+                    cls = f"{obj.__class__.__module__}.{obj.__class__.__name__} "
+                    cls = cls.removeprefix("PyQt5.")
+                    cls = cls.removeprefix("las_dispersion_scan.widgets.")
```
Comment (klauer) [inline pcdsutils/qt/designer_display.py:67]:
```suggestion
```

**Review by klauer — APPROVED** [review]
LGTM 👍

File: pcdsutils/qt/designer_display.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsutils/qt/designer_display.py:67]: haha good catch

**Comment by ZLLentz** [comment]
pypi build fails with
```
ERROR: pytest-qt requires either PySide2, PySide6, PyQt5 or PyQt6 installed.
```
I think I need to add an extras something something environment variable for the ci

**Comment by klauer** [comment]
https://github.com/pcdshub/typhos/blob/14af093e703baccfe4e18c2eef5d394d9213c8fe/.travis.yml#L13
and
https://github.com/pcdshub/typhos/blob/14af093e703baccfe4e18c2eef5d394d9213c8fe/.travis.yml#L19

**Comment by ZLLentz** [comment]
thank you I was about to run it without CONDA_EXTRAS

File: pcdsutils/qt/designer_display.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if not hasattr(cls, "ui_form"):
+            try:
+                path = str(Path(cls.ui_dir) / cls.filename)
+            except AttributeError:
+                path = str(cls.filename)
+            cls.ui_form, _ = loadUiType(path)
+
+    def __init__(self, *args, **kwargs):
+        """Apply the file to this widget when the instance is created"""
+        self._load_ui_if_needed()
```
Comment (tangkong) [inline pcdsutils/qt/designer_display.py:42]: in atef we call this with an `__init_subclass__` hook, which seems at first glance to be more or less the same as this?  (loading the ui file if it's not been already?)

I'm fine with it either way, but was curious about the change.

**Review by tangkong — APPROVED** [review]
LGTM, I really liked this when I first used it, and imagine it'll be very useful moving forward

File: pcdsutils/qt/designer_display.py (hunk 2 above)
Comment (ZLLentz) [reply pcdsutils/qt/designer_display.py:42]: I think this is so we don't have to have the ui directory defined at import time for our subdisplays

Comment (klauer) [reply pcdsutils/qt/designer_display.py:42]: My logic for this route was that on-import `.ui` compiling may slow down applications that don't actually use the displays. In this case, there's only a loading penalty on first usage instead.

**Pull request merged**